controllers/performanceprofile_controller.go

```diff
@@ -156,6 +175,36 @@ func (r *PerformanceProfileReconciler) mcpToPerformanceProfile(mcpObj client.Obj
 	return requests
 }
 
+func (r *PerformanceProfileReconciler) tunedProfileToPerformanceProfile(tunedProfileObj client.Object) []reconcile.Request {
+	node := &corev1.Node{}
+	key := types.NamespacedName{
+		// the tuned profile name is the same as node
+		Name: tunedProfileObj.GetName(),
+	}
+
+	if err := r.Get(context.TODO(), key, node); err != nil {
+		klog.Errorf("failed to get the tuned profile %+v", key)
+		return nil
+	}
+
+	profiles := &performancev2.PerformanceProfileList{}
+	if err := r.List(context.TODO(), profiles); err != nil {
+		klog.Error("failed to get performance profiles")
+		return nil
+	}
+
+	var requests []reconcile.Request
+	for i, profile := range profiles.Items {
+		profileNodeSelector := labels.Set(profile.Spec.NodeSelector)
+		nodeLabels := labels.Set(node.Labels)
```

Comment:
Can we have a situation here where we dont have a 1:1 match ? 
If Matches does not do so, is a contains labels check more suitable here ?

Reply:
For sure we have such a situation, the node can have a lot of different labels but we should match one that is specified under the node selector?

Reply:
if this is the functionality here then we are good.  just make sure its not the other way around